core/src/main/java/org/apache/carbondata/core/cache/dictionary/ReverseDictionaryCache.java

```diff
@@ -167,12 +167,9 @@ private Dictionary getDictionary(
       DictionaryColumnUniqueIdentifier dictionaryColumnUniqueIdentifier)
       throws CarbonUtilException {
     Dictionary reverseDictionary = null;
-    // create column dictionary info object only if dictionary and its
-    // metadata file exists for a given column identifier
-    if (!isFileExistsForGivenColumn(dictionaryColumnUniqueIdentifier)) {
-      throw new CarbonUtilException(
-          "Either dictionary or its metadata does not exist for column identifier :: "
-              + dictionaryColumnUniqueIdentifier.getColumnIdentifier());
+    // create column dictionary info object only if it is primitive type.
+    if (dictionaryColumnUniqueIdentifier.getDataType().isComplexType()) {
+      return null;
```

Comment:
I am not sure this condition check is right or not? Maybe we still need to do dictionary encoding for complex type?
@QiangCai  pls review this logic